src/incubator/calendar/index.tsx

```diff
@@ -31,23 +31,23 @@ function Calendar(props: PropsWithChildren<CalendarProps>) {
     showExtraDays = true
   } = props;
 
-  const [items] = useState<DateObjectWithOptionalDay[]>(() =>
+  const [monthItems] = useState<DateObjectWithOptionalDay[]>(() =>
     generateMonthItems(initialDate, YEARS_RANGE, YEARS_RANGE));
 
   const getItemIndex = useCallback((date: number) => {
     'worklet';
     const dateObject = getDateObject(date);
-    for (let i = 0; i < items.length; i++) {
-      if (items[i].month === dateObject.month && items[i].year === dateObject.year) {
+    for (let i = 0; i < monthItems.length; i++) {
+      if (monthItems[i].month === dateObject.month && monthItems[i].year === dateObject.year) {
         return i;
       }
     }
     return -1;
   },
-  [items]);
+  [monthItems]);
 
   const flashListRef = useRef();
-  const current = useSharedValue<number>(initialDate);
+  const current = useSharedValue<number>(new Date(initialDate).setHours(0, 0, 0, 0));
```

Comment:
You're changing the `initialDate` the user provided, We can probably set 0 to `sec` \ `ms` but I'm not sure about `min` and `hours`

Reply:
The time of the date is not important as we are working with days, we need the timestamp to react to day changes and not to hours/mins/secs..